File: Programmers/118667.py
from collections import deque

def solution(queue1, queue2):
    queue1 = deque(queue1)
    queue2 = deque(queue2)
    sum1 = sum(queue1)
    sum2 = sum(queue2)
    l = len(queue1)

    answer = 0
    while True:
        if sum1 > sum2:
            sum1 -= queue1[0]
            sum2 += queue1[0]
            queue2.append(queue1.popleft())
        elif sum1 < sum2:
            sum2 -= queue2[0]
            sum1 += queue2[0]
            queue1.append(queue2.popleft())
        else:
            return answer
        answer += 1
        if answer >= l * 4:
            return -1
    

File: Programmers/118667.js
function solution(queue1, queue2) {
  let sum1 = queue1.reduce((a, b) => a + b, 0);
  let sum2 = queue2.reduce((a, b) => a + b, 0);
  const half = (sum1 + sum2) / 2;
  const q = [...queue1, ...queue2];
  const n = queue1.length;
  let pointer1 = 0;
  let pointer2 = n;

  for (let count = 0; count < n * 3; count++) {
    if (sum1 === half) return count;
    else if (sum1 > half) {
      sum1 -= q[pointer1];
      pointer1++;
    } else {
      sum1 += q[pointer2];
      pointer2++;
    }
  }

  return -1;
}
